simulator/risc_v/riscv_instr.cpp

```diff
@@ -122,99 +159,168 @@ struct RISCVTableEntry
     Src2 src2;
     Dst  dst;
     uint32 mem_size;
+    base_t base = RV_ANY;
     RISCVTableEntry() = delete;
+    bool check_print_dst( Reg reg) const
+    {
+        return ( reg != Reg::ZERO)
+            && ( reg != Reg::MEPC)
+            && ( reg != Reg::SEPC);
+    }
+    bool check_print_src( Reg reg) const
+    {
+        return ( reg != Reg::ZERO)
+            && ( reg != Reg::SEPC)
+            && ( reg != Reg::MEPC)
+            && ( reg != Reg::CSR);
+    }
 };
 
 template<typename I>
 static const std::vector<RISCVTableEntry<I>> cmd_desc =
 {
     /*-------------- I --------------*/
-    {'I', instr_lui,   execute_lui<I>,   OUT_ARITHM, 'U', Imm::LOGIC, Src1::ZERO, Src2::ZERO, Dst::RD, 0},
-    {'I', instr_auipc, execute_auipc<I>, OUT_ARITHM, 'U', Imm::LOGIC, Src1::ZERO, Src2::ZERO, Dst::RD, 0},
+    {'I', instr_lui,    execute_lui<I>,    OUT_ARITHM, 'U', Imm::LOGIC, Src1::ZERO, Src2::ZERO, Dst::RD,   0},
+    {'I', instr_auipc,  execute_auipc<I>,  OUT_ARITHM, 'U', Imm::LOGIC, Src1::ZERO, Src2::ZERO, Dst::RD,   0},
     // Jumps and branches
-    {'I', instr_jal,   execute_jal<I>,   OUT_J_JUMP, 'J', Imm::ARITH, Src1::ZERO, Src2::ZERO, Dst::RD, 0},
-    {'I', instr_jalr,  execute_jalr<I>,  OUT_R_JUMP, 'I', Imm::LOGIC, Src1::RS1,  Src2::ZERO, Dst::RD, 0},
-    {'I', instr_beq,   execute_beq<I>,   OUT_BRANCH, 'B', Imm::ARITH, Src1::RS1,  Src2::RS2,  Dst::ZERO, 0},
-    {'I', instr_bne,   execute_bne<I>,   OUT_BRANCH, 'B', Imm::ARITH, Src1::RS1,  Src2::RS2,  Dst::ZERO, 0},
-    {'I', instr_blt,   execute_blt<I>,   OUT_BRANCH, 'B', Imm::ARITH, Src1::RS1,  Src2::RS2,  Dst::ZERO, 0},
-    {'I', instr_bge,   execute_bge<I>,   OUT_BRANCH, 'B', Imm::ARITH, Src1::RS1,  Src2::RS2,  Dst::ZERO, 0},
-    {'I', instr_bltu,  execute_bltu<I>,  OUT_BRANCH, 'B', Imm::ARITH, Src1::RS1,  Src2::RS2,  Dst::ZERO, 0},
-    {'I', instr_bgeu,  execute_bgeu<I>,  OUT_BRANCH, 'B', Imm::ARITH, Src1::RS1,  Src2::RS2,  Dst::ZERO, 0},
+    {'I', instr_jal,    execute_jal<I>,    OUT_J_JUMP, 'J', Imm::ARITH, Src1::ZERO, Src2::ZERO, Dst::RD,   0},
```

Comment:
Please do not attempt to make code "prettier". It will introduce merge conflicts with other people, while benefits are very small.